src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -355,7 +375,158 @@ private string ResolveTargetPath(string rootPath, FileContainerItem item, string
             return absolutePath;
         }
 
-        private List<FileContainerItem> GetFilteredItems(List<FileContainerItem> items, IEnumerable<Func<string, bool>> minimatchFuncs)
+        public List<FileContainerItem> GetFilteredItems(List<FileContainerItem> items, string[] minimatchPatterns, Options customMinimatchOptions)
```

Comment:
Could we split this function into several ones?

Reply:
Done. Expanding of braces in patterns, updating of patterns map and applying patterns map to list of items were moved to separate functions.